src/Settings.ts

```diff
@@ -40,6 +40,6 @@ export namespace Settings {
     }
 
     function _getMavenSection<T>(section: string, resource?: Uri): T {
-        return workspace.getConfiguration("maven", resource).get<T>(section);
+        return workspace.getConfiguration("maven", resource || null).get<T>(section);

```

Comment:
Make the default value of resource to `null` instead of making it as an optional parameter?

Reply:
`export function getConfiguration(section?: string, resource?: Uri)` 
 if resource is `undefined`, it gets warning. It suggests you to explicitly pass a `null` if you need to get conf for any resource.

Reply:
And if you think this is wired, we can keep it as optional parameter, and ensure to pass a `null` when calling it.